drafts/liberty-dev-mode-gradle.adoc

```diff
@@ -0,0 +1,128 @@
+
+== Automatically recompiling and deploying to the running server
+
+Once dev mode is started, source code changes are automatically compiled and deployed to the running server. If any features are added to the `server.xml` file, they are installed automatically.
+
+For example, adding the the `mpHealth-2.1` feature to server.xml to enable health checks will make the health endpoint accessible at http://localhost:9080/health.
+
+
+image::/img/blog/dev_mode_server_xml.png[mp-health Server XML]
+image::/img/blog/dev_mode_health_check_1.png[MicroProfile Health check]
+
+When you add code into your application for liveness checks, you can immediately see the updated result when you access the URL again.
+
+[source,java]
+----
+package io.openliberty.sample.system;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import java.lang.management.MemoryMXBean;
+import java.lang.management.ManagementFactory;
+
+import org.eclipse.microprofile.health.Liveness;
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+
+@Liveness
+@ApplicationScoped
+public class SystemLivenessCheck implements HealthCheck {
+
+    @Override
+    public HealthCheckResponse call() {
+        MemoryMXBean memBean = ManagementFactory.getMemoryMXBean();
+        long memUsed = memBean.getHeapMemoryUsage().getUsed();
+        long memMax = memBean.getHeapMemoryUsage().getMax();
+  
+        return HealthCheckResponse.named(
+            SystemResource.class.getSimpleName() + " liveness check")
+                                  .withData("memory used", memUsed)
+                                  .withData("memory max", memMax)
+                                  .state(memUsed < memMax * 0.9).build();
+    }
+    
+}
+----
+
+image::/img/blog/dev_mode_health_check_2.png[MicroProfile Health check]
+
+== Running tests automatically and on demand
+
+Go to the console where you started dev mode, and press Enter. This will run the tests on a separate thread while dev mode is still active.
+
+To run tests automatically after every change, include the `--hotTests` parameter when starting dev mode.  For example: `gradle libertyDev --hotTests`
+
+== Attaching a debugger
+
+At any point during dev mode, you can attach a Java debugger to debug your application.  By default, the debug port is `7777`, but another port can be used and shown in the console output if the default port is not available.
+
+In your SystemLivenessCheck Class, set a breakpoint inside the call() method.
+
+Attach your IDE's debugger to port 7777.
+For example, in VS Code, click Debug > Add Configuration... > Java: Attach > set "port": "7777". Then View > Debug > select Debug (Attach), then press the arrow icon to start debugging.
+
+In your browser, go to http://localhost:9080/health.
+
+Notice your IDE pauses at the breakpoint that you set, allowing you to debug.
```

Comment:
Instead of providing instructions here, also rephrase this as an example scenario and provide a screenshot if appropriate.  And leave out the VS Code specific steps but just make it more general e.g. "In the example below, we attach our IDE's debugger to port 7777"